python/revng/api/manager.py

```diff
@@ -418,6 +439,8 @@ def set_input(self, container_name: str, content: Union[bytes, str], _key=None)
                 f"Failed loading user provided input for container {container_name}"
             )
 
+        self.save()
```

Comment:
`self.save` simply tells the manager to serialize its state in the working directory.

Reply:
Why not save only `begin` given we have the granularity to say so?

Reply:
It could be done, but it could cause problems because it's the (C++) manager that decides the directory layout for saving and respects that layout when loading back the data; so there would be 2 places where saving happens that would need to be kept in sync.

Reply:
Reworked saving functionality a bit more, now only the `begin` step is saved when setting the input